ios/FluentUI/Drawer/DrawerController.swift

```diff
@@ -606,7 +624,8 @@ open class DrawerController: UIViewController {
     private func updatePreferredContentSize(isExpanded: Bool) {
         isPreferredContentSizeBeingChangedInternally = true
         if isExpanded {
-            preferredContentSize.height = UIScreen.main.bounds.height
+            let expandedContentHeight = (preferredMaximumExpansionHeight < UIScreen.main.bounds.height && preferredMaximumExpansionHeight >= originalDrawerHeight) ? preferredMaximumExpansionHeight : UIScreen.main.bounds.height
```

Comment:
it seems like preferredMaximumExpansionHeight == -1 is checked multiple times? that might be something we want to cache? then in case it is -1, we could skip checking "preferredMaximumExpansionHeight < UIScreen.main.bounds.height && preferredMaximumExpansionHeight >= originalDrawerHeight" and just use the screen height...

Reply:
Sorry for missing this in the first go. Broken down the if- else cascade to avoid any unnecessary preferredMaximumExpansionHeight == -1 checks. In the latest commit, there is only one usage of preferredMaximumExpansionHeight == -1, so not caching it.